### DISCUSSION TITLE
Installation conflicts for mayavi

### DISCUSSION THREAD

**Discussion by Srinivas-Ganti** [NanoComp/meep#2131] [Q&A]
Hello,

Does anyone know a tried and tested way to get everything installed and setup correctly? 

I want to be able to do all the tutorials so I can eventually try out my own simulations based on the things I learn from the docs page. Would really appreciate everyone for your suggestions! So far, I tried to follow the installation as given on the docs page on my Linux (Ubuntu) machine - first I created a new anaconda env (python 3.7), I then installed pymeep and pymeep extras using 

conda create -n mp -c conda-forge pymeep pymeep-extras

This completed successfully and I am able to run the tutorial examples (at the [Transmission Spectrum](https://meep.readthedocs.io/en/latest/Python_Tutorials/Resonant_Modes_and_Transmission_in_a_Waveguide_Cavity/) one at the moment).

I then tried to install mayavi using their conda to prepare for the 3D visualisation parts shown in the later tutorials using conda

conda install -c anaconda mayavi

This installation doesnt complete and it seems there are many conflicting packages. I attach the output from the terminal, I get unsatisfiable error.

[terminalOutput_myaviInstall.txt](https://github.com/NanoComp/meep/files/9113969/terminalOutput_myaviInstall.txt)
.

**Comment by Srinivas-Ganti** [comment]
Update: Managed to complete the [visualisation example](https://meep.readthedocs.io/en/latest/Python_Tutorials/Basics/#visualizing-3d-structures) with Paraview after converting the numpy.ndarray data to vtk using the h5tovtk utility outside my notebook environment. Still no luck with Mayavi however. 

![image](https://user-images.githubusercontent.com/64644132/179073588-d7264e7e-8d93-4c8c-88b1-54d1c99bbb33.png)

**Comment by tummasn** [comment]
I recently installed Mayavi and am using it for Meep. 
I have taken the following steps:
* new python environment
* install vkt
* install mayavi==4.7.2
* install Meep
* install PyQt

Hope it helps :)

**Reply by Srinivas-Ganti** [reply]
Hi, thanks for this info. Please could you also specify which version of python you installed in the new environment?

**Reply by tummasn** [reply]
3.9.13

**Reply by Srinivas-Ganti** [reply]
this didnt work for me unfortunately. I am unable to complete the mayavi install, I get this error:


ResolvePackageNotFound: 
  - python=3.1
